src/main/java/com/linkedin/kmf/services/ConsumeService.java

```diff
@@ -110,7 +106,8 @@ public ConsumeService(Map<String, Object> props, String name, CompletableFuture<
     if (props.containsKey(ConsumeServiceConfig.CONSUMER_PROPS_CONFIG)) {
       props.forEach(consumerProps::putIfAbsent);
     }
-    _consumer = (KMBaseConsumer) Class.forName(consumerClassName).getConstructor(String.class, Properties.class).newInstance(topic, consumerProps);
+    _consumer = (KMBaseConsumer) Class.forName(consumerClassName).getConstructor(String.class, Properties.class).newInstance(_topic, consumerProps);
```

Comment:
Because the consume service is creating it's own consumer rather than using an injected consumer it's not possible to mock the consumer without lots of effort.  This is one of the challenges with unit testing this class.  If the consume service either takes a consumer or  consumer factory as a parameter then mocking becomes possible, which means unit testing becomes possible.

Reply:
In order to render unit testing possible for `ConsumeService`, I will make the `consumeService` take in consumer / consumer factory as an argument in a follow PR if that's feasible and okay with you.